tests/add_subclass_to_py.rs

```diff
@@ -0,0 +1,20 @@
+//! See https://github.com/PyO3/pyo3/issues/4452.
+
+use pyo3::prelude::*;
+
+#[pyclass(subclass)]
+struct BaseClass {}
+
+#[pyclass(extends=BaseClass)]
+struct SubClass {
+    _data: i32,
+}
+
+#[test]
+#[should_panic]
+fn add_subclass_to_py_is_unsound() {
+    Python::with_gil(|py| {
+        let base = Py::new(py, BaseClass {}).unwrap();
+        let _subclass = PyClassInitializer::from(base).add_subclass(SubClass { _data: 42 });
+    });
+}
```

Comment:
IMO this could just be a unit test in `pyclass_init.rs`, doesn't really need a new file?

Reply:
This does not compile... Declaring a `#[pyclass]` in the same crate does not work.

Reply:
If I follow what you're saying I think the trick might be to use `#[pyclass(crate = crate)]` (like other unit tests do).

Reply:
I was sure I tried that and it didn't work, but it works :confused: